src/components/confirm/confirm.test.js

```diff
@@ -0,0 +1,386 @@
+
+      dialogPreview().trigger("keydown", keyCode("Esc"));
+
+      dialogPreview().should("be.visible");
+    });
+
+    it("should render Confirm with aria-label prop", () => {
+      CypressMountWithProviders(<ConfirmComponent aria-label="cypress-aria" />);
+
+      dialogPreview().should("have.attr", "aria-label", "cypress-aria");
+    });
+
+    it("should check confirm button has save icon", () => {
+      CypressMountWithProviders(
+        <ConfirmComponent confirmButtonIconType="Save" />
+      );
+
+      getDataElementByValue("Save").should("have.attr", "type", "Save");
+    });
+
+    it("should check cancel button has bin icon", () => {
+      CypressMountWithProviders(
+        <ConfirmComponent cancelButtonIconType="bin" />
+      );
+
+      getDataElementByValue("bin").should("have.attr", "type", "bin");
+    });
+
+    it("should render Confirm with aria-labelledby prop", () => {
+      CypressMountWithProviders(
+        <Confirm open aria-labelledby={testPropValue} />
+      );
+
+      dialogPreview().should("have.attr", "aria-labelledby", testPropValue);
+    });
+
+    it("should render Confirm with aria-describedby prop", () => {
+      CypressMountWithProviders(
+        <Confirm open aria-describedby={testPropValue} />
+      );
+
+      dialogPreview().should("have.attr", "aria-describedby", testPropValue);
+    });
+
+    it("should focus first element", () => {
+      CypressMountWithProviders(<ConfirmComponent />);
+      const firstElement = () => cy.get('[data-element="first-element"]');
+      firstElement().should("be.focused");
+    });
+
+    it.each([
+      ["before", 0],
+      ["after", 1],
+    ])(
+      "should render Confirm using focus first element",
+      (position, positionInElement) => {
+        CypressMountWithProviders(
+          <ConfirmComponent
+            cancelButtonIconType="bin"
+            cancelButtonIconPosition={position}
+          />
+        );
+
+        cancelButton()
+          .children()
+          .eq(positionInElement)
+          .should("have.attr", "data-element", "bin");
+      }
+    );
+  });
+
+  describe("should render Confirm component for event tests", () => {
+    let callback;
+
+    beforeEach(() => {
+      callback = cy.stub();
+    });
+
+    it("should call onCancel callback when a click event is triggered", () => {
+      CypressMountWithProviders(<ConfirmComponent onCancel={callback} />);
+
+      cancelButton()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call onConfirm callback when a click event is triggered", () => {
+      CypressMountWithProviders(<ConfirmComponent onConfirm={callback} />);
+
+      confirmButton()
+        .click()
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
```

Comment:
You could add the escape key test here if you wanted, I'm not sure I've got the syntax right but something similar to below:

```
 CypressMountWithProviders(<ConfirmComponent onCancel={callback} />);

  dialogPreview().trigger("keyup", keyCode("Esc")).then(() => {
          // eslint-disable-next-line no-unused-expressions
          expect(callback).to.have.been.calledOnce;
        });
      
```